src/shell/shell_misc.cpp

```diff
@@ -420,9 +420,12 @@ void DOS_Shell::InputCommand(char * line) {
 	if (l_completion.size()) l_completion.clear();
 
 	/* DOS %variable% substitution */
-	const auto dos = static_cast<Section_prop *>(control->GetSection("dos"));
-	assert(dos);
-	if (dos->Get_bool("expand_shell_variable"))
+	const auto dos_section = static_cast<Section_prop *>(control->GetSection("dos"));
+	assert(dos_section);
+	if (strcmp(dos_section->Get_string("expand_shell_variable"), "true") == 0)
+		ProcessCmdLineEnvVarStitution(line);
+	else if (strcmp(dos_section->Get_string("expand_shell_variable"), "auto") == 0 
+					&& dos.version.major >= 7)
```

Comment:
Suggest assigning the `dos_section->Get_string("expand_shell_variable")` to a temporary string_view:

```c++
std::string_view expand_shell_variable_pref = 
                 dos_section->Get_string("expand_shell_variable");
```

This will allow for natural comparisons:

``` c++
if (expand_shell_variable_pref == "true") {
   ...
} else if (expand_shell_variable_pref == "auto"
           && dos.version.major >= 7) {
   ...
}
```

Reply:
Last point - it would be nice to move this cong-setting-checking up to the point where the `[dos]` section is parsed (so it's done only once, and saved in a static bool, like `should_expand_variables`, which we could use here instead). 

Because this a lot of repeat logic to run through for every `%variable%` the shell encounters.  (I'm not not worried about performance, more about putting things where they logically belong).

That would be a very nice refactoring, if you can do it!

Reply:
I'm not quite sure where to move the check to. The other changes were made.

Reply:
Can wait for a future refactoring; thanks @rderooy !